fs/operations.go

```diff
@@ -1303,3 +1364,24 @@ func MoveFile(fdst Fs, fsrc Fs, dstFileName string, srcFileName string) (err err
 func CopyFile(fdst Fs, fsrc Fs, dstFileName string, srcFileName string) (err error) {
 	return moveOrCopyFile(fdst, fsrc, dstFileName, srcFileName, true)
 }
+
+func (j jsonPrinter) Start(w io.Writer) {
+	syncFprintf(w, "[")
+}
+
+func (j jsonPrinter) End(w io.Writer) {
+	syncFprintf(w, "]")
+}
+
+func (j *jsonPrinter) Fprintf(w io.Writer, skipSeperator bool, format string, a ...interface{}) {
+	// Lets skip the seperator when calling the function itself
+	if j.seperatorRequired && !skipSeperator {
+		j.Fprintf(w, true, ",")
+	}
+	// Set the seperator before locking so parallel things don't get in the way of the lock
+	j.seperatorRequired = true
+
+	j.mu.Lock()
```

Comment:
Did move the lock and the defer unlock to the top?

When you say locked up - is it deadlocked?  If so send the process a SIGQUIT or type CTRL-\ and it will show backtraces of all the running goroutines which should give you a hint.

I can't think why it would lock up!